packages/terra-date-time-picker/src/DateTimeUtils.js

```diff
@@ -75,37 +75,58 @@ class DateTimeUtils {
    * Synchronize only the time for a given moment object.
    * @param {object} momentDate - The moment object to synchronize the date and time.
    * @param {string} time - The time to synchronize with the moment object.
+   * @param {boolean} hasSeconds - If true seconds will be processed
    * @return {object} - The synchronized moment object.
    */
-  static updateTime(momentDate, time) {
+  static updateTime(momentDate, time, hasSeconds) {
     if (!momentDate || !momentDate.isValid()) {
       return null;
     }
 
     const newDate = momentDate.clone();
-    const date = moment(time, 'HH:mm', true);
+    const timeFormat = hasSeconds ? 'HH:mm:ss' : 'HH:mm';
+    const date = moment(time, timeFormat, true);
+
+    if (hasSeconds) {
+      return newDate.hour(date.get('hour')).minute(date.get('minute')).second(date.get('second'));
+    }
 
     return newDate.hour(date.get('hour')).minute(date.get('minute'));
   }
 
+  /**
+   * Gets the time from a date and time
+   * @param {string} time An ISO 8601 string to get the time of
+   * @param {boolean} hasSeconds Whether or not seconds should be retrieved
+   * @return {string} The time from the date and time string
+   */
+  static getTime(time, hasSeconds) {
+    const timeFormat = hasSeconds ? 'HH:mm:ss' : 'HH:mm';
+    return DateUtil.formatISODate(time, timeFormat);
+  }
+
   /**
    * Determines if the date and time are valid and conform to the given format.
    * @param {string} date - The date to validate.
    * @param {string} time - The time to validate.
    * @param {string} format - The date/time format to use for the validation.
+   * @param {boolean} hasSeconds Whether or not the time should consider having seconds valid
    * @return {boolean} - True if both the date and time are valid and conform to the format.
    */
-  static isValidDateTime(date, time, format) {
-    return DateUtil.isValidDate(date, format) && DateTimeUtils.isValidTime(time);
+  static isValidDateTime(date, time, format, hasSeconds) {
```

Comment:
There are two places [[1](https://github.com/cerner/terra-framework/blob/master/packages/terra-date-time-picker/src/DateTimePicker.jsx#L200)] [[2](https://github.com/cerner/terra-framework/blob/master/packages/terra-date-time-picker/src/DateTimePicker.jsx#L215)] in DateTimePicker.jsx that calls isValidDateTime. The `showSeconds` prop needs to be passed down when making those calls. Otherwise, the ambiguous hour logic won't work when the second input field is shown.

Reply:
Fixed in https://github.com/cerner/terra-framework/pull/722/commits/f0b8c47950943f019c0502eebc2439ea2f2f0921 and changes to the other file were reverted in https://github.com/cerner/terra-framework/pull/722/commits/7695bd247be326a7c12526d12767b514d75660e1